src/sql/workbench/browser/modelComponents/button.component.ts

```diff
@@ -24,33 +24,44 @@ import { convertSize } from 'sql/base/browser/dom';
 @Component({
 	selector: 'modelview-button',
 	template: `
+	<p>BT: {{this._buttonType}}</p>
 	<div>
 		<label for={{this.label}}>
 			<div #input style="width: 100%">
 				<input #fileInput *ngIf="this.isFile === true" id={{this.label}} type="file" accept="{{ this.fileType }}" style="display: none">
 			</div>
 		</label>
 	</div>
+
 	`
 })
+
 export default class ButtonComponent extends ComponentWithIconBase implements IComponent, OnDestroy, AfterViewInit {
 	@Input() descriptor: IComponentDescriptor;
 	@Input() modelStore: IModelStore;
 	private _button: Button;
+	public _buttonType = <azdata.ButtonType>'Normal';
```

Comment:
why not just use azdata.ButtonType.Normal

Reply:
Because azdata doesn't exist in core at runtime so that would fail. You can only reference the types in core - not any actual implementations.

If this enum was converted to a const enum that would probably also work I guess. Although the fact that that's not done anywhere else makes me wonder if there's a good reason for that...

Typically this is handled by just redefining the enum in the core code - but there's an issue there with where we had it defined (sqlExtHostTypes) with the import rules that I pinged Anthony about.